README.md

```diff
@@ -1,2 +1,76 @@
+- Number of accounts for a swap is a limiter for some routers. Manifest swaps that do not use global orders achieve the theoretical minimum number of accounts.
+- CU is a major cost for market makers. Benchmarking shows Manifest has improved higher percentiles which is what the large spenders really care about.
+- Silent failures are an unfortunate feature request from market makers because of how solana transactions work. Manifest rejects this at the infra level, but still allows those who need it to handle in the wrapper.
+- Token 22 is the new version of token program. While it is not useful for defi and will make orderbooks less efficient, there are some notable tokens that will use it. Manifest only takes the performance hit to support token22 precisely when needed and moving token22 tokens, and only then.
+
+## Design Overview
+### Data Structure
+
+The innovation that allows this next leap in onchain trading is the hypertree [`hypertree`](https://github.com/CKS-Systems/manifest/tree/main/lib). All data in the market account fits into graph nodes of the same size (80 bytes), which lets independent data structures grow without being fully initialized from the start by interleaving
+
+The market account holds all relevant information. It begins with a header that stores all of the fixed information for the market like BaseMint, QuoteMint. All variable data (RestingOrders and ClaimedSeats) are in the dynamic
+byte array after the header. There are 3 RedBlack trees for Bids, Asks,
+ClaimedSeats and 1 LinkedList for FreeListNodes, overlapping across each other. All are graphs where each vertex along with adjacency list fits in 80 bytes, allowing them to use the same blocks.
+
+<pre>
+--------------------------------------------------------------------------------------------------------
+|                   Header                    |                               Dynamic                   |
+--------------------------------------------------------------------------------------------------------
+| BaseMint, QuoteMint, BidsRootIndex, ...     | Bid | Ask | FreeListNode | Seat | Seat | Bid | Bid | Ask|
+--------------------------------------------------------------------------------------------------------
+</pre>
+
+### Core vs Wrapper
+Manifest views the orderbook as an infrastructure layer primitive. Other orderbooks get bogged down by special feature requests from trading teams that ultimately make the program bloated and confusing. Manifest strives to only include features that are absolutely necessary to be in the base layer. Anything that can be handled at layers above on the stack will not be done in manifest. This simplification makes formal verification of the program feasible.
```

Comment:
Manifest implements the orderbook as an infrastructure layer primitive and creates the purest form of risk exchange possible.